sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubBufferedProducerAsyncClient.java

```diff
@@ -0,0 +1,334 @@
+     *
+     * @param partitionId The unique identifier of a partition associated with the Event Hub.
+     *
+     * @return the quantity of events for the specific partition.
+     */
+    public int getBufferedEventCount(String partitionId) {
+        return partitionBatchMap.get(partitionId).size();
+    }
+
+    /**
+     * Enqueue a single event without send options, will be sent to event hub by default send options.
+     *
+     * @param eventData The {@link EventData} to store in client temporarily to send to event hub.
+     * @throws AmqpException if the size of {@code events} exceed the maximum size of a single batch.
+     * @return Gets a {@link Mono} that completes when eventData enqueued.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> enqueueEvent(EventData eventData) {
+        return enqueueEvent(eventData, DEFAULT_SEND_OPTIONS);
+    }
+
+    /**
+     * Enqueue a single event with send options.
+     *
+     * @param eventData The {@link EventData} will be stored in client temporarily and sent to event hub.
+     * @param options A set of options used to configure the {@link EventDataBatch}.
+     * @throws AmqpException if the size of {@code events} exceed the maximum size of a single batch.
+     * @return Gets a {@link Mono} that completes when eventData enqueued.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> enqueueEvent(EventData eventData, SendOptions options) {
+        List<EventData> list = new ArrayList<>();
+        list.add(eventData);
+        return enqueueEvents(list, options);
+    }
+
+    /**
+     * Enqueue iterable events without send options, will be sent to event hub by default send options.
+     *
+     * @param events Events will be stored in client temporarily and sent to event hub.
+     * @throws AmqpException if the size of {@code events} exceed the maximum size of a single batch.
+     * @return Gets a {@link Mono} that completes when eventData enqueued.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> enqueueEvents(Iterable<EventData> events) {
+        return enqueueEvents(events, DEFAULT_SEND_OPTIONS);
+    }
+
+    /**
+     * Enqueue iterable events with send options.
+     *
+     * @param events Events will be stored in client temporarily and sent to event hub.
+     * @param options A set of options used to configure the {@link EventDataBatch}.
+     * @throws AmqpException if the size of {@code events} exceed the maximum size of a single batch.
+     * @return Gets a {@link Mono} that completes when eventData enqueued.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> enqueueEvents(Iterable<EventData> events, SendOptions options) {
+        Objects.requireNonNull(events, "'eventData' cannot be null.");
+        ArrayList<EventData> list = new ArrayList<>();
+        events.forEach(list::add);
+        addBatchToPartitionMap(options.getPartitionId(), list);
```

Comment:
This should be asynchronous because addBatchToPartition could take a long time, so users aren't benefitting from the async framework.

Ideally, you'd update `addBatchToPartitionMap` from `void` to `Mono<Void>`